test/libsolidity/syntaxTests/abiEncoder/encodeCallMismatch.sol

```diff
@@ -0,0 +1,45 @@
+
+contract C {
+	function f(int a) public {}
+	function f2(int a, string memory b) public {}
+	function f3(int a, int b) public {}
+
+	function fail() public returns(bytes memory) {
```

Comment:
More syntax test ideas:
- Function pointer from:
    - Internal function
    - Internal function variable
    - Library function
    - Interface function
- Function arguments:
   - `()`
   - `[]`
   - `(x,)`
   - `((x, y), (z, w))`
   - Result of a function call that returns multiple values
   - Value returned by a getter of a public state variable of struct type (it unpacks the struct into individual components)